FreshRSS/FreshRSS
# Add ability to bypass image lazy loading for networks

Eagerly-loaded images are great for the local network, it feels awesome when web comic pages can be opened within a fraction of a second.

However, eagerly-loaded images suck when you're on a phone with cellular data connectivity. The front page loading can take multiple seconds.

This change adds a way to disable lazy loading for certain IP ranges (your home network).

How to test the feature manually:

1. Enable image lazy loading, and test that images are indeed loaded lazily. The easiest way is to check the browser's network monitoring console.
2. Add your local network (e.g. 10.0.0.0/8 or 192.168.0.0/16) to the exclusion list (Settings -> Reading -> Misc).
3. Check that images are loaded eagerly.

Pull request checklist:


PS: I'm sorry, I'm not very familiar with PHP and my code might not be well-organized.
